Makefile

```diff
@@ -633,6 +633,19 @@ generate-nomad-rust-client: ## Generate the Nomad rust client from OpenAPI
 # should use that here.
 	$(MAKE) format-prettier
 
```

Comment:
1. Could we get a quick README or something on how to use and consume this?
2. Not necessarily for _this_ PR but would it make sense to exercise this in CI as a "hey here's proof the jaeger setup all still works"?

Reply:
Also, in general, it seems odd to me that this is a
1. enable
2. do things
3. disable

and not a 
1. WITH_JAEGER_TRACING=1 do things

wish we had context managers or RAII in bash...

Reply:
I had a similar thought... I wonder if we could have a one-time "setup" target that creates the tracing-enabled builder, and then a `WITH_TRACING` macro / flag / etc. that would add the `--builder=XXX` option to the `docker buildx bake` invocations.

``` make
ifdef WITH_TRACING
buildx_builder_args := --builder=grapl-tracing
endif

DOCKER_BUILDX_BAKE_HCL := docker buildx bake --file=docker-bake.hcl $(buildx_builder_args)
```
Then you can do `WITH_TRACING=1 make local-infrastructure` to get tracing enabled (given the builder exists, of course).

(If there was anything else to add tracing to, we could stick that behind the `WITH_TRACING` variable, too.)

It's still not airtight, though, because it still relies on a) having the builder set up and b) that Jaeger is running.

Make is kinda weird for this 🤷 

(These are just thoughts... either way is fine; I'm thinking out loud here)

Reply:
Taking another look, I think the existing code conflates the _creation_ of a builder with the _using_ of that builder. In particular, disabling the builder does not also delete that builder (I don't think that it _should_ do that, just to be clear), so I think we'll have an error the next time we run `make buildx-tracing` because it will try to create another builder with the same name as one that already exists.

Decoupling the creation of the builder from the using or not-using of that builder seems like a worthwhile change.

Reply:
1. I've added a README in the docs/development folder
2. I've updated the code to use an environment variable and a one-time setup command